lib/src/compiler.ts

```diff
@@ -0,0 +1,210 @@
+// Copyright 2023 Google LLC. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import {Observable} from 'rxjs';
+
+import * as p from 'path';
+import * as supportsColor from 'supports-color';
+import {deprotofySourceSpan} from './deprotofy-span';
+import {Dispatcher, DispatcherHandlers} from './dispatcher';
+import {Exception} from './exception';
+import {ImporterRegistry} from './importer-registry';
+import {
+  legacyImporterProtocol,
+  removeLegacyImporter,
+  removeLegacyImporterFromSpan,
+} from './legacy/utils';
+import {MessageTransformer} from './message-transformer';
+import {PacketTransformer} from './packet-transformer';
+import * as utils from './utils';
+import * as proto from './vendor/embedded_sass_pb';
+import {SourceSpan} from './vendor/sass';
+import {CompileResult} from './vendor/sass/compile';
+import {Options, StringOptions} from './vendor/sass/options';
+
+/// Allow the legacy API to pass in an option signaling to the modern API that
+/// it's being run in legacy mode.
+///
+/// This is not intended for API users to pass in, and may be broken without
+/// warning in the future.
+export type OptionsWithLegacy<sync extends 'sync' | 'async'> = Options<sync> & {
+  legacy?: boolean;
+};
+
+/// Allow the legacy API to pass in an option signaling to the modern API that
+/// it's being run in legacy mode.
+///
+/// This is not intended for API users to pass in, and may be broken without
+/// warning in the future.
+export type StringOptionsWithLegacy<sync extends 'sync' | 'async'> =
+  StringOptions<sync> & {legacy?: boolean};
+
+/**
+ * Creates a dispatcher that dispatches messages from the given `stdout` stream.
+ */
+export function createDispatcher<sync extends 'sync' | 'async'>(
+  stdout: Observable<Buffer>,
+  writeStdin: (buffer: Buffer) => void,
+  handlers: DispatcherHandlers<sync>
+): Dispatcher<sync> {
+  const packetTransformer = new PacketTransformer(stdout, writeStdin);
+
+  const messageTransformer = new MessageTransformer(
+    packetTransformer.outboundProtobufs$,
+    packet => packetTransformer.writeInboundProtobuf(packet)
+  );
+
+  return new Dispatcher<sync>(
+    // Since we only use one compilation per process, we can get away with
+    // hardcoding a compilation ID. Once we support multiple concurrent
+    // compilations with the same process, we'll need to ensure each one uses a
+    // unique ID.
```

Comment:
This is no longer true. Now that there are multiple compilations per process, you'll need to multiplex the requests and responses for each compilation to the appropriate dispatcher.

Reply:
Added in fbbe407b79adc730bdf1e6a2be11b0b4464d79b6